### PR TITLE
CIF-2732 - Teaser: add option to open CTA in a new Tab

### PR DESCRIPTION
**Pull request by LSantha**
* added v3 of commerce teaser with support for link target
 * added Java and JS unit tests
 * updated examples library
 * updated component readme
 * deprecated v1 and v2 of commerce teaser

## Related Issue

CIF-2732

## How Has This Been Tested?

Manually, unit test.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes and the overall coverage did not decrease.
- [X] All unit tests pass on CircleCi.
- [X] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v3/teaser/CommerceTeaserImpl.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                if (StringUtils.isNotBlank(categoryUid)) {
+                    actionUrl = urlProvider.toCategoryUrl(request, categoryPage, categoryUid);
+                    identifier = new CommerceIdentifierImpl(categoryUid, CommerceIdentifier.IdentifierType.UID,
+                        CommerceIdentifier.EntityType.CATEGORY);
+                } else if (StringUtils.isNotBlank(productSku)) {
+                    actionUrl = urlProvider.toProductUrl(request, productPage, productSku);
+                    identifier = new CommerceIdentifierImpl(productSku, CommerceIdentifier.IdentifierType.SKU,
+                        CommerceIdentifier.EntityType.PRODUCT);
+                } else if (StringUtils.isNotBlank(link)) {
+                    actionUrl = link + ".html";
```
Comment (Buuhuu) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v3/teaser/CommerceTeaserImpl.java:111]: Could we use the action from the WCM Teaser instead? This implementation applies some additional logic on the link which is private and we cannot easily replicate it (Calling LinkHandler#getLink())

**Review by herzog31 — APPROVED** [review]
LGTM

Could you please add to all the `@Deprecated` annotations (except for tests) corresponding JavaDoc which explains why it was deprecated and what should be used instead?

**Comment by LSantha** [comment]
I had to increase the aem-uber.jar dependency to 6.5.8 which contains ComponentStyleInfo.

**Comment by LSantha** [comment]
I've added deprecation information too.

**Pull request merged**